src/lib/logical_query_plan/compare_lqps.cpp

```diff
@@ -0,0 +1,291 @@
+#include "compare_lqps.hpp"
+
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "logical_query_plan/aggregate_node.hpp"
+#include "logical_query_plan/create_view_node.hpp"
+#include "logical_query_plan/delete_node.hpp"
+#include "logical_query_plan/drop_view_node.hpp"
+#include "logical_query_plan/dummy_table_node.hpp"
+#include "logical_query_plan/insert_node.hpp"
+#include "logical_query_plan/join_node.hpp"
+#include "logical_query_plan/limit_node.hpp"
+#include "logical_query_plan/logical_plan_root_node.hpp"
+#include "logical_query_plan/mock_node.hpp"
+#include "logical_query_plan/predicate_node.hpp"
+#include "logical_query_plan/projection_node.hpp"
+#include "logical_query_plan/show_columns_node.hpp"
+#include "logical_query_plan/show_tables_node.hpp"
+#include "logical_query_plan/sort_node.hpp"
+#include "logical_query_plan/stored_table_node.hpp"
+#include "logical_query_plan/union_node.hpp"
+#include "logical_query_plan/update_node.hpp"
+#include "logical_query_plan/validate_node.hpp"
+
+namespace opossum {
+
+SemanticLQPCompare::SemanticLQPCompare(const std::shared_ptr<const AbstractLQPNode>& lhs, const std::shared_ptr<const AbstractLQPNode>& rhs): _lhs(lhs), _rhs(rhs) {
```

Comment:
Couple of thoughts about this (that I'll split in multiple comments to that they are easier to follow):
a) Is this really an object that can be constructed? Can't we do this statically?

Reply:
I have this tendency to put more complex algorithms, especially this with that have local state shared by multiple sub-algorithms into classes to facilitate tracking this state. In this case SemanticLQPCompare tracks the mis-matching sub-trees.
Depending on where we place to _are_semantically_equal functions, this class could well be turned into a function, I think